Tactical/Weapons.cpp

```diff
@@ -2719,32 +2723,36 @@ BOOLEAN UseGunNCTH( SOLDIERTYPE *pSoldier , INT32 sTargetGridNo )
 	{
 		if ( WillExplosiveWeaponFail( pSoldier, pObjHand ) )
 		{
+			INT16 sX, sY;
+			ConvertGridNoToCenterCellXY(pSoldier->sGridNo, &sX, &sY);

```

Comment:
IgniteExplosion and AddItemToPool have INT32 sGridNo as a parameter and at that point the functions have their own private copies of pSoldier->sGridNo instead of a reference, right? 

Other functions, at least in this case do not use pSoldier->sGridNo at all, but good point. I should probably check if an unintended change can happen elsewhere where I did a similar thing.